viofs/svc/virtiofs.cpp

```diff
@@ -486,39 +489,91 @@ static DWORD FindDeviceInterface(PHANDLE Device)
     return ERROR_SUCCESS;
 }
 
-static VOID UpdateLocalUidGid(VIRTFS *VirtFs, DWORD SessionId)
+static NTSTATUS FspToolGetTokenInfo(HANDLE Token,
+    TOKEN_INFORMATION_CLASS TokenInformationClass, PVOID *PInfo)
 {
-    PWSTR UserName = NULL;
-    LPUSER_INFO_3 UserInfo = NULL;
-    DWORD BytesReturned;
-    NET_API_STATUS Status;
-    BOOL Result;
+   PVOID Info{ NULL };
+   DWORD Size;
+   NTSTATUS Result;
 
-    Result = WTSQuerySessionInformation(WTS_CURRENT_SERVER_HANDLE, SessionId,
-        WTSUserName, &UserName, &BytesReturned);
+   if (GetTokenInformation(Token, TokenInformationClass, 0, 0, &Size))
+   {
+      Result = STATUS_INVALID_PARAMETER;
+      goto exit;
+   }
 
-    if (Result == TRUE)
-    {
-        Status = NetUserGetInfo(NULL, UserName, 3, (LPBYTE *)&UserInfo);
+   if (ERROR_INSUFFICIENT_BUFFER != GetLastError())
+   {
+      Result = FspNtStatusFromWin32(GetLastError());
+      goto exit;
+   }
 
-        if (Status == NERR_Success)
-        {
-            // Use an account from local machine's user DB as the file's
-            // owner (0x30000 + RID).
-            VirtFs->LocalUid = UserInfo->usri3_user_id + 0x30000;
-            VirtFs->LocalGid = UserInfo->usri3_primary_group_id + 0x30000;
-        }
+   Info = HeapAlloc(GetProcessHeap(), 0, Size);
+   if (NULL == Info)
+   {
+      Result = STATUS_INSUFFICIENT_RESOURCES;
+      goto exit;
+   }
+
+   if (!GetTokenInformation(Token, TokenInformationClass, Info, Size, &Size))
+   {
+      Result = FspNtStatusFromWin32(GetLastError());
+      goto exit;
+   }
+
+   *PInfo = Info;
+   Result = STATUS_SUCCESS;
 
-        if (UserInfo != NULL)
+exit:
+   if (!NT_SUCCESS(Result))
+      SafeHeapFree(Info);
+
+   return Result;
+}
```

Comment:
This function still has 3 spaces as indentation width

Reply:
Ah, the function I copied from winfsp. The horror. Only think worse is 2 spaces ;) On it.

Reply:
@YanVugenfirer I haven't marked as resolved mostly because of this